style_guides/js_style_guide.md

```diff
@@ -346,90 +476,67 @@ if (a == '') {
 
 And **never** use multiple ternaries together
 
-*Right:*
-
 ```js
-var foo = (a === b) ? 1 : 2;
-```
-
-*Wrong:*
+// good
+const foo = (a === b) ? 1 : 2;
 
-```js
-var foo = (a === b) ? 1 : (a === c) ? 2 : 3;
+// bad
+const foo = (a === b) ? 1 : (a === c) ? 2 : 3;
 ```
 
 ## Do not extend built-in prototypes
 
 Do not extend the prototype of native JavaScript objects. Your future self will
 be forever grateful.
 
-*Right:*
-
 ```js
-var a = [];
-if (!a.length) {
-  console.log('winning');
-}
-```
-
-*Wrong:*
-
-```js
-Array.prototype.empty = function() {
+// bad
+Array.prototype.empty = function () {
   return !this.length;
 }
-
-var a = [];
-if (a.empty()) {
-  console.log('losing');
-}
 ```
 
 ## Use descriptive conditions
 
-Any non-trivial conditions should be assigned to a descriptively named variables, broken into
-several names variables, or converted to be a function:
-
-*Right:*
+Any non-trivial conditions should be converted to functions or assigned to
+descriptively named variables.
 
 ```js
-var thing = ...;
-var isShape = thing instanceof Shape;
-var notSquare = !(thing instanceof Square);
-var largerThan10 = isShape && thing.size > 10;
-
-if (isShape && notSquare && largerThan10) {
-  console.log('some big polygon');
+// best
+function isShape(thing) {
+  return thing instanceof Shape;
+}
+function notSquare(thing) {
+  return !(thing instanceof Square);
+}
+if (isShape(thing) && notSquare(thing)) {
+  ...
 }
-```
 
-*Wrong:*
+// good
+const isShape = thing instanceof Shape;
+const notSquare = !(thing instanceof Square);
+if (isShape && notSquare) {
+  ...
+}
 
-```js
-if (
-  thing instanceof Shape
-  && !(thing instanceof Square)
-  && thing.size > 10
-) {
-  console.log('bigger than ten?? Woah!');
+// bad
+if (thing instanceof Shape && !(thing instanceof Square)) {
+  ...
 }
 ```
 
 ## Name regular expressions
 
-*Right:*
-
 ```js
-var validPasswordRE = /^(?=.*\d).{4,}$/;
+// good
+const validPasswordRE = /^(?=.*\d).{4,}$/;
 
 if (password.length >= 4 && validPasswordRE.test(password)) {
```

Comment:
This seems to break the rule about using variable names for comparison, or using a const for conditionals with an if.